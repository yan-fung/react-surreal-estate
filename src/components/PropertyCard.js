import React from "react";
import Proptypes from "prop-types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faBed,
  faBath,
  faSterlingSign,
  faEnvelope,
  faLocationDot,
} from "@fortawesome/free-solid-svg-icons";
import "../styles/property-card.css";
import logo from "../styles/images/homelogo.png";

const PropertyCard = ({
  title,
  type,
  bathrooms,
  bedrooms,
  price,
  city,
  email,
}) => {
  return (
    <div className="property-card">
      <div className="property-card-container">
        <div className="property-card__logo">
          <img src={logo} alt="logo" width="30px" height="25px" />
        </div>
        <div className="property-card__title">{title}</div>
        <div className="property-card__type__city">
          <FontAwesomeIcon icon={faLocationDot} className="icon-location" />
          {type} - {city}
        </div>
        <div
          className="property-card_bathrooms"
          data-testid="property-bathrooms"
        >
          <FontAwesomeIcon icon={faBath} className="icon" />
          Bathrooms: {bathrooms}
        </div>
        <div className="property-card_bedrooms">
          <FontAwesomeIcon icon={faBed} className="icon" />
          Bedrooms: {bedrooms}
        </div>
        <div className="property-card_price">
          <FontAwesomeIcon icon={faSterlingSign} className="icon-price" />
          {price}
        </div>
        <div className="property-card__email">
          <a href={`mailto:${email}`} className="email">
            <FontAwesomeIcon icon={faEnvelope} className="icon-email" />
            Email
          </a>
        </div>
      </div>
    </div>
  );
};

export default PropertyCard;

PropertyCard.propTypes = {
  title: Proptypes.string.isRequired,
  type: Proptypes.string.isRequired,
  bathrooms: Proptypes.string.isRequired,
  bedrooms: Proptypes.string.isRequired,
  price: Proptypes.string.isRequired,
  city: Proptypes.string.isRequired,
  email: Proptypes.string.isRequired,
};
