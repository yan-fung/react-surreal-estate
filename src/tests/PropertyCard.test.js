import React from "react";
import { render } from "@testing-library/react";
import PropertyCard from "../components/PropertyCard";

describe("PropertyCard", () => {
  const validProps = {
    title: "test_title",
    type: "test type",
    bathrooms: "test bathrooms",
    bedrooms: "test bedrooms",
    price: "test price",
    city: "test city",
    email: "test email",
  };
  it("renders correctly", () => {
    const { asFragment } = render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct value for props", () => {
    const { getByTestId, getByText } = render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />
    );
    expect(getByText(/test_title/)).toHaveClass("property-card__title");
    expect(getByTestId("property-bathrooms")).toHaveAttribute(
      "class",
      "property-card_bathrooms"
    );
  });
});
