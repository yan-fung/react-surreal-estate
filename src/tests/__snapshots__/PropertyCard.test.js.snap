// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <div
      class="property-card-container"
    >
      <div
        class="property-card__logo"
      >
        <img
          alt="logo"
          height="25px"
          src="homelogo.png"
          width="30px"
        />
      </div>
      <div
        class="property-card__title"
      >
        test_title
      </div>
      <div
        class="property-card__type__city"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-location-dot icon-location"
          data-icon="location-dot"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M215.7 499.2C267 435 384 279.4 384 192C384 86 298 0 192 0S0 86 0 192c0 87.4 117 243 168.3 307.2c12.3 15.3 35.1 15.3 47.4 0zM192 128a64 64 0 1 1 0 128 64 64 0 1 1 0-128z"
            fill="currentColor"
          />
        </svg>
        test type - test city
      </div>
      <div
        class="property-card_bathrooms"
        data-testid="property-bathrooms"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-bath icon"
          data-icon="bath"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M96 77.3c0-7.3 5.9-13.3 13.3-13.3c3.5 0 6.9 1.4 9.4 3.9l14.9 14.9C130 91.8 128 101.7 128 112c0 19.9 7.2 38 19.2 52c-5.3 9.2-4 21.1 3.8 29c9.4 9.4 24.6 9.4 33.9 0L289 89c9.4-9.4 9.4-24.6 0-33.9c-7.8-7.9-19.8-9.1-29-3.8C246 39.2 227.9 32 208 32c-10.3 0-20.2 2-29.2 5.5L163.9 22.6C149.4 8.1 129.7 0 109.3 0C66.6 0 32 34.6 32 77.3V256c-17.7 0-32 14.3-32 32s14.3 32 32 32H480c17.7 0 32-14.3 32-32s-14.3-32-32-32H96V77.3zM32 352v16c0 28.4 12.4 54 32 71.6V480c0 17.7 14.3 32 32 32s32-14.3 32-32V464H384v16c0 17.7 14.3 32 32 32s32-14.3 32-32V439.6c19.6-17.6 32-43.1 32-71.6V352H32z"
            fill="currentColor"
          />
        </svg>
        Bathrooms: test bathrooms
      </div>
      <div
        class="property-card_bedrooms"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-bed icon"
          data-icon="bed"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M32 32c17.7 0 32 14.3 32 32V320H288V160c0-17.7 14.3-32 32-32H544c53 0 96 43 96 96V448c0 17.7-14.3 32-32 32s-32-14.3-32-32V416H352 320 64v32c0 17.7-14.3 32-32 32s-32-14.3-32-32V64C0 46.3 14.3 32 32 32zm144 96a80 80 0 1 1 0 160 80 80 0 1 1 0-160z"
            fill="currentColor"
          />
        </svg>
        Bedrooms: test bedrooms
      </div>
      <div
        class="property-card_price"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-sterling-sign icon-price"
          data-icon="sterling-sign"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M112 160.4c0-35.5 28.8-64.4 64.4-64.4c6.9 0 13.8 1.1 20.4 3.3l81.2 27.1c16.8 5.6 34.9-3.5 40.5-20.2s-3.5-34.9-20.2-40.5L217 38.6c-13.1-4.4-26.8-6.6-40.6-6.6C105.5 32 48 89.5 48 160.4V224H32c-17.7 0-32 14.3-32 32s14.3 32 32 32H48v44.5c0 17.4-4.7 34.5-13.7 49.4L4.6 431.5c-5.9 9.9-6.1 22.2-.4 32.2S20.5 480 32 480H288c17.7 0 32-14.3 32-32s-14.3-32-32-32H88.5l.7-1.1C104.1 390 112 361.5 112 332.5V288H224c17.7 0 32-14.3 32-32s-14.3-32-32-32H112V160.4z"
            fill="currentColor"
          />
        </svg>
        test price
      </div>
      <div
        class="property-card__email"
      >
        <a
          class="email"
          href="mailto:test email"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-envelope icon-email"
            data-icon="envelope"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4L236.8 313.6c11.4 8.5 27 8.5 38.4 0L492.8 150.4c12.1-9.1 19.2-23.3 19.2-38.4c0-26.5-21.5-48-48-48H48zM0 176V384c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V176L294.4 339.2c-22.8 17.1-54 17.1-76.8 0L0 176z"
              fill="currentColor"
            />
          </svg>
          Email
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
